api/src/main/java/io/grpc/CallCredentials.java

```diff
@@ -62,6 +62,23 @@ public abstract void applyRequestMetadata(
   @ExperimentalApi("https://github.com/grpc/grpc-java/issues/1914")
   public abstract void thisUsesUnstableApi();
 
+  /**
+   * Determine if the security level on the transport is higher than or equal to the minimum
+   * security level required by {@link CallCredentials} in order to transfer it.
+   *
+   * <p>It should be called in {@link applyRequestMetadata} before sending any
+   * individual RPC. The RPC should not be sent if the API returns false.
+   * More details can be found in https://github.com/grpc/proposal/pull/167.
+   *
+   * @param requestInfo request-related information
+   * @param minSecurity minimum security level required by {@link CallCredentials}
+   */
+  public static final boolean allowedSecurityLevel(
```

Comment:
Thinking about it more, let's make this protected. Only extending classes will need to call this. It wouldn't really hurt anything by having it public, but pollutes what consumers of CallCredentials would see.